### DISCUSSION TITLE
Django and Etherium

### DISCUSSION THREAD

**Discussion by drbazpython** [smartcontractkit/full-blockchain-solidity-course-py#228] [General]
Fantastic course. Thanks Patrick. I'm about half way through and enjoying every transaction and contract.
Just thinking about adding a UI via Django.

**Comment by PatrickAlphaC** [comment]
Boom! Glad you're enjoying it. 

Yeah that would be awesome to do a django project!!! If you make one, we could add it as a brownie mix!

**Reply by drbazpython** [reply]
Couldn't integrate brownie into a Django project. Resorted to using subprocess.call to run the "brownie run deploy.py" script.
Then I realised that brownie was available as a python library ... doh! Read the docs! 
So tomorrow I will try again ... onwards and upwards

**Comment by Verde888** [comment]
This is a great idea and a path I want to follow, having not much experience myself with front-end development. Don't forget to show us your project!

**Comment by drbazpython** [comment]
Project progressing well, if slowly. Unit testing functions to deploy and interact with contracts via brownie api within a Django app. Almost TDD !!!!! 
Basic Django UI to display deployment and transaction details.  Using the SimpleStorage.sol contract as a basic example and deploying to Ganache and Rinkeby.

**Reply by cromewar** [reply]
never seen a django implementation with smart contracts, do you have a github repo with this project? I would like to see how you did it.

**Reply by drbazpython** [reply]
No repo yet. Still getting to grips with testing the brownie api inside a Django app. 
Once I'm happy with the testing, I'll create a simple UI with Django.

**Comment by drbazpython** [comment]
It's the brownie api that has required a lot of learning. 
It will be relatively easy to use the Django template engine to present contract data to the UI

**Reply by Verde888** [reply]
Hello Dr Smith,

how is it going? I'm really looking forward to the results of your work! :)
I think having the possibility to make a simple interface in Django would open many possibilities for those among us who have no idea about writing front-end in javascript (which is one of the reasons to learn from this course and not many others that involve web3.js...)

Bests,

**Comment by drbazpython** [comment]
A quick update - I have decided to move away from a Brownie-Django project as I found some difficulties with Brownie and its api.  The Brownie CLI is great for learning about blockchain etc so I recommend using Brownie to gain an understanding of the technology without the need to "develop" code.
I am preparing a very simple demo of using Django with Web3.py. This type of integration should be within the reach of anyone who has a basic understanding of Django. So rather than wait for my demo, I recommend that you learn Django and web3.py. There are lots of Django tutorials on YouTube, and I recommend looking at the videos from Coding Entrepreneurs. The building of a Django-Web3 project would be a good way to learn two technologies, without relying on other people.

**Reply by Verde888** [reply]
Hi, thanks for the update.
I'm familiar with Django. My limited understanding of Web.py comes from this very smart contracts course - as you probably know it switched to Brownie after presenting the basics of Web3.py. I was interested to see how to integrate my new "blockchain" knowledge into my web development. I understand that JS and React are pretty much standard for front-end nowadays, however it would be nice to have a workaround to provide some basic but functional frontend with django :)
